integration/src/integration/java/pl/allegro/tech/hermes/integration/FrontendAuthenticationConfigurationTest.java

```diff
@@ -104,13 +105,49 @@ public void shouldNotAuthenticateUserWithInvalidCredentials() throws Throwable {
     @Test
     public void shouldNotAuthenticateUserWithoutCredentials() throws Throwable {
         //when
-        Response response = publisher.publish("someGroup.topicWithAuthorization",
-                TestMessage.of("hello", "world").body());
+        Response response = publisher.publish("someGroup.topicWithAuthorization", MESSAGE);
 
         //then
         assertThat(response.getStatus()).isEqualTo(StatusCodes.UNAUTHORIZED);
     }
 
+    @Test
+    public void shouldPublishOnTopicWithGrantedPermission() {
+        //given
+        Topic topic = TopicBuilder.topic("g.granted")
+                .withPublisher(USERNAME)
+                .withAuthEnabled()
+                .withAuthEnforced()
+                .build();
+
+        operations.buildTopic(topic);
+        Map<String, String> headers = getHeadersWithAuthentication(USERNAME, PASSWORD);
+
+        //when
+        Response response = publisher.publish(topic.getQualifiedName(), MESSAGE, headers);
+
+        //then
+        assertThat(response.getStatusInfo().getFamily()).isEqualTo(SUCCESSFUL);
+    }
+
+    @Test
+    public void shouldNotPublishOnTopicWithoutGrantedPermission() {
```

Comment:
IMO it's worth spawning another frontend :) Having tests for `pro_active` mode is a plus btw. 👍